actionpack/lib/action_controller/metal/implicit_render.rb

```diff
@@ -2,14 +2,26 @@ module ActionController
   module ImplicitRender
     def send_action(method, *args)
       ret = super
-      default_render unless performed?
-      ret
+      return ret if performed?
+
+      lookup_context.variants = request.variant
+
+      if template_exists?(method, _prefixes)
+        default_render
+      else
+        default_head
+      end
     end
 
     def default_render(*args)
       render(*args)
     end
 
+    def default_head(*)
```

Comment:
We should extend `method_for_action` to use this too, right? Otherwise we should just put this back in the else clause above.